htdocs/core/lib/functions.lib.php

```diff
@@ -4863,7 +4863,7 @@ function price2num($amount, $rounding = '', $option = 0)
 	if ($option != 1) {	// If not a PHP number or unknown, we change or clean format
 		//print 'PP'.$amount.' - '.$dec.' - '.$thousand.' - '.intval($amount).'<br>';
 		if (!is_numeric($amount)) {
-			$amount = preg_replace('/[a-zA-Z\/\\\*\(\)\<\>]/', '', $amount);
+			$amount = preg_replace('/[a-zA-Z\/\\\*\(\)\<\>\_\-]/', '', $amount);
```

Comment:
The - must not be removed because it is also used as a sign.
A fix would be to protect the code entered when creating a vat rate to make the - forbidden into code

Reply:
Ok, @eldy you are obviously right for -
But I am afraid, in any case, that my correction (and even corrected here by you) was not able to cover the case of languages with non-Latin characters (what I saw recently [here for example)](https://www.dolibarr.org/forum/t/issue-with-vat-in-vendor-invoice/20093/8) about cyrillic.
Don't you think it would be better to first remove everything from and after the 1st non-numeric character (<>0-9, .,-) from the left of the string?